Add Rank Points Per Match to Rankings Table

maherb opened the pull request:
added rankings_enhanced that uses model {"team_number": {"ranking_score_per_match": number}}

towards #1572

phil-lopreiato commented on models/event.py:
```diff
@@ -214,6 +215,23 @@ def rankings(self):
+    def rankings_enhanced(self):
+        if self._rankings is not None and self.year == 2016:
+            self._rankings_enhanced = {}
+            team_number_index = self._rankings[0].index("Team")
```
use `self.rankings` for these. If `self.rankings` hasn't been called yet, then `self._rankings` will be `None` since it hasn't been lazy loaded yet

maherb replied:
Functionality implemented, screenshot attached. I see two problems with the current implementation:
1) Doesn't look great
2) Not possible to sort by average RS, which seems to be a big part of why users want this

I missed Phil's comment there. I will include that correction in the next commit.

![image](https://cloud.githubusercontent.com/assets/16950452/17161953/886dc250-537f-11e6-8770-19720a33886c.png)

phil-lopreiato commented:
I'm a fan!

> 2) Not possible to sort by average RS, which seems to be a big part of why users want this

How about we make average RS its own column? These tables have a cool trick where you can click on the header to sort the values. You can [insert](http://www.tutorialspoint.com/python/list_insert.htm) values into the `full_rankings` rows. Be sure to insert a header in the first row and a blank value for rows where no average is available, to keep the columns lined up.

fangeugene commented on templates_jinja2/event_details.html:
```diff
@@ -211,6 +211,7 @@ <h3 class="panel-title">Event in progress</h3>
               {% endfor %}
             </tbody>
           </table>
+          {% if event.year in [2012, 2013, 2014, 2016] %}<p>*This column is calculated for your convenience by The Blue Alliance using data from FIRST and is not official.</p>{% endif %}
```
italicize FIRST

maherb commented:
Added consts.ranking_indexes to all manipulating rankings from all years, and added the <ranking criterion>/match column to all events 2007 onward except 2015, which already ranks by average
![image](https://cloud.githubusercontent.com/assets/16950452/17163597/ea94d568-538e-11e6-8aec-00639a7d9ee3.png)

fangeugene commented:
Did you commit ranking_indexes?

phil-lopreiato commented on controllers/event_controller.py:
```diff
@@ -170,6 +171,21 @@ def _render(self, event_key):
+        # rankings processing for ranking score per match
+        full_rankings = event.rankings
+        rankings_enhanced = event.rankings_enhanced
+        if rankings_enhanced is not None:
```
add `and event.year in RankingIndexes.CUMULATIVE_RANKING_SCORE` so we don't get crashes when 2017 events are added before the constants are updated

phil-lopreiato commented:
LGTM! 👍
